index.js

```diff
@@ -85,20 +85,103 @@ module.exports = {
           .catch(this._errorMessage.bind(this));
       },
 
-      activate(/* context */) {
+      activate(context) {
         const revisionKey = this.readConfig('revisionKey');
 
         this.log(`preparing to activate ${revisionKey}`, {
           verbose: true
         });
 
-        return this._uploadDeployInfo(this.s3)
-          .then(() => {
-            this.log(`✔  activated revison ${revisionKey}`, {
-              verbose: true
-            });
-          })
-          .catch(this._errorMessage.bind(this));
+        let _this = this;
+        return this.fetchRevisions(context).then(function(revisions) {
+           let found = revisions.revisions.map(function(element) { return element.revision; }).indexOf(revisionKey);
+           if (found >= 0) {
+
+             return _this._uploadDeployInfo(_this.s3)
+               .then(() => {
+                 _this.log(`✔  activated revison ${revisionKey}`, {
+                   verbose: true
+                 });
+               })
+               .catch(_this._errorMessage.bind(_this));
+
+           } else {
+             return RSVP.reject("REVISION NOT FOUND!"); // see how we should handle a pipeline failure
+           }
+         });
+      },
+
+      fetchRevisions: function(context) {
+        return this._list(context)
+          .then(function(revisions) {
+            return {
+              revisions: revisions
+            };
+          });
+      },
+
+      _list: function(/* context */) {
+        let bucket      = this.readConfig('bucket');
+        let deployArchive      = this.readConfig('deployArchive');
+        let revisionPrefix = deployArchive + "-";
```

Comment:
This causes an error if a `prefix` is used (ie. putting the files in a directory in the bucket). Could you please add the prefix value to the `revisionPrefix` if it exists? This was done elsewhere with a ternary.